lib/react_on_rails/version_checker.rb

```diff
@@ -12,25 +12,28 @@ def self.build
       new(NodePackageVersion.build)
     end
 
+    def self.instance(package_json_path)
+      new(NodePackageVersion.new(package_json_path))
+    end
+
     def initialize(node_package_version)
       @node_package_version = node_package_version
     end
 
     # For compatibility, the gem and the node package versions should always match,
     # unless the user really knows what they're doing. So we will give a
     # warning if they do not.
-    def raise_if_gem_and_node_package_versions_differ
-      return if node_package_version.relative_path?
+    def log_if_gem_and_node_package_versions_differ
+      return if node_package_version.raw.nil? || node_package_version.relative_path?
```

Comment:
`return if node_package_version.raw.nil? || node_package_version.relative_path?`

why?

Reply:
Because `raw` is `nil` if the package.json file doesn't exist.

Do you want us to create a warning for that?

Reply:
I just modified the code so that `raw` is nil if package.json or the react-on-rails dependency in the package.json is missing.